src/main/kotlin/pl/droidsonroids/gradle/jacoco/testkit/JacocoTestKitExtension.kt

```diff
@@ -5,28 +5,36 @@ import org.gradle.api.Task
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.TaskProvider
 import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
+import org.gradle.testing.jacoco.tasks.JacocoReport
 import java.io.File
 
 open class JacocoTestKitExtension(private val project: Project) {
     val jacocoRuntimePathProvider: Provider<String> = project.provider { project.configurations.getByName(Configurations.jacocoRuntime).asPath }
 
     fun applyTo(configurationRuntime: String, taskProvider: TaskProvider<Task>) {
         with(project) {
+            val destinationFile = taskProvider.map { task ->
+                val extension = task.extensions.getByType(JacocoTaskExtension::class.java)
+                val file = checkNotNull(extension.destinationFile) { "destinationFile is missing on task $name jacoco extension" }
+
+                File(file.parentFile, "${file.nameWithoutExtension}-gradle-runner.${file.extension}")
+            }
+
             val jacocoTestKitPropertiesTask = tasks.register(
                 generatePropertiesTaskName(taskProvider.name),
                 GenerateJaCoCoTestKitProperties::class.java
             ) {
                 it.outputFile = File(buildDir, "testkit/${taskProvider.name}/testkit-gradle.properties")
-                it.destinationFile.set(
-                    taskProvider.map {
-                        task -> task.extensions.getByType(JacocoTaskExtension::class.java).destinationFile!!.path
-                    }.get()
-                )
+                it.destinationFile.set(destinationFile.get().path)
                 it.jacocoRuntimePath.set(jacocoRuntimePathProvider)
             }
 
             dependencies.add(configurationRuntime, files(testKitDir(taskProvider.name)))
             taskProvider.configure { it.dependsOn(jacocoTestKitPropertiesTask) }
+
+            tasks.withType(JacocoReport::class.java)
+                .matching { it.name == "jacoco${taskProvider.name.capitalize()}Report" }
```

Comment:
If provider name starts with `i` and default locale at runtime is set to Turkish or Azeri then capitalization will result in capital dotted `İ` so no match will be probably found.
Anyway we have a similar existing capitalize https://github.com/koral--/jacoco-gradle-testkit-plugin/pull/15#discussion_r418568736 so it's fine for now to leave it also here.

Reply:
Actually this was the most controversial part of the change, as I generally don't like reference tasks by name miming their internal plugin logic to name them. But I couldn't find a way to archive this without making too much refactor on the current code.